### ISSUE TITLE
Visualization sleuther should find non-time-series CSVs that are graphable

### ISSUE THREAD

**Issue by AlexGilleran** [TerriaJS/magda#670]
Right now there are a lot of datasets that would make perfectly reasonable bar graphs that don't show visualisations because they're not time-series. We should change that. Given that you can select axes in vega lite it shouldn't be too much of a problem to just let the user choose what they want to graph.

**Comment by steve9164** [comment]
Sorta blocked by #585 (it'd make sense to fix visualisation logic before going further on this).

**Comment by t83714** [comment]
@AlexGilleran @tkeuneman @steve9164 
I think we might need a generic story to make sure every CSV file (no mater includes time series or not) can be visualized by our program:

Every CSV file can be considered as a set of multi-dimension data. Our problem of having a generic ( and `reasonable`) solution is to pick relevant dimensions to x & y-axis depends on the nature of the different chart types. Once we solve this, all remaining dimensions can put into tooltip of plot points.

We probably can implement a solution in this way:

-  We consider each column of the CSV file as one dimension. The dimension name is the column name of the column. We assume the first row always contains column names.
- Our program needs to be able to categorise the dimensions to the following types by their data:
  - Numeric: In order to make more sense of the data, we need put them into more category in future. **But not now**:
    - Ratio or Float
    - Currency
    - Unit or Int
   - Discrete value (string value that cannot be converted as time or numeric)  
   - Time / Date (Temporal)
- We always use X axis as 
  - category axis i.e.  only accepts discrete value type dimensions
  - Or time / date axis i.e. Only accepts Time / Date type dimensions
  - If Time / Date type dimensions present, by default, we should use X axis as the time axis
- We always use Y axis as the value axis
  - and it accepts only Numeric type dimensions
  - If there is no Numeric type dimensions in the CSV file, we should
    - allow `Discrete value` type or `Time / Date` type dimensions to be assigned to Y axis by count() aggregation. i.e. Treat all values of the dimension as discrete string values and then count the number of their appearance as data to be displayed on Y-axis

Therefore, we can process any CSV files in the following steps:
- Try to determine the data type of the columns (this has been done by Visual Sleuther but we can do some improvement on frontend)
- Determin a column for x-axis:
    - If there is one Time/Data type column, the first Time/Data type column will be assigned to X-axis. 
    - Otherwise, the first `Discrete value` type column will be assigned to X-axis
    - If no neither Time/Data type column nor `Discrete value` type column, the first Numeric column will be used
- Determin a column for y-axis:
    - If there is one Numeric type column, the first Numeric type column will be used.
    - Otherwise, the first `Discrete value` type column will be used by count() aggregation (see above)
    - If No `Discrete value` type column, the first `Date / Time` type column will be used by count() aggregation (see above)
- If the CSV file has only one column, an extra column will be generated and filled with the first column heading as values.
- We will allow users to pick the column / dimension by themselves.
  - Only Time/Data type or `Discrete value` type columns will be shown as available options for x-axis unless there is no such type (in this case, all columns will be available for selection)
  - Only Numeric value type columns will be shown as available options for Y-axis. unless there is no such type (in this case, any columns selected will be applied to y-axis by count() aggregation)
  - At this moment, only allow on column / dimension selected for x-axis or y-axis

We will only support the following chart types:
- Bar or Columns
- Lines or Area
- Pie (not dounut support)
- Buble or scatter

Let's me know your ideas regarding this story. We also can talk about this in details.
